FV3/atmos_cubed_sphere/model/dyn_core.F90

```diff
@@ -922,29 +938,29 @@ subroutine dyn_core(npx, npy, npz, ng, sphum, nq, bdt, n_split, zvir, cp, akap,
     enddo           ! end openMP k-loop
     !$ser savepoint D_SW-Out
     !$ser data delpcd=vt delpd=delp ptcd=ptc ptd=pt ud=u vd=v wd=w ucd=uc vcd=vc uad=ua vad=va divgdd=divgd mfxd=mfx mfyd=mfy cxd=cx cyd=cy crxd=crx cryd=cry xfxd=xfx yfxd=yfx q_cond=q_con heat_sourced=heat_source diss_estd=diss_est nord_vd=nord_v damp_vtd=damp_vt
-    
+    !$ser verbatim call finalize_kbuff()
     if (flagstruct%regional) then
        call exch_uv(domain, bd, npz, vc, uc)
        call exch_uv(domain, bd, npz, u,  v )
     endif
                                                      call timing_off('d_sw')
 
     if( flagstruct%fill_dp ) call mix_dp(hydrostatic, w, delp, pt, npz, ak, bk, .false., flagstruct%fv_debug, bd)
+  
+    !$ser savepoint HaloUpdate-In 
+    !$ser data complete=complete_halo_false  array=delp rank=tile_id
     !$ser savepoint HaloUpdate-In
-    !$ser data delp=delp pt=pt q_con=q_con
+    !$ser data complete=complete_halo_true array=pt rank=tile_id 
```

Comment:
Are npx, npy, npz being output here?

`tile_id` isn't necessarily the same as rank, is it? When there are more than 6 ranks?

Reply:
npx, npy, and npz will be part of the grid savepoint that gets created in atmosphere.F90 -- you'll have access to it by using the global  'grid' object. If you consider the 'tile' to be the cube face, then it's not the same, but I was thinking of the tile as the domain the rank is operating on. But, the semantics are confusing. This variable is indeed the rank gotten from call mpi_comm_rank(MPI_COMM_WORLD, tile_id,ier), rather than the domain%tile_id, so I will change the name of this variable to rank.

Reply:
For now I can work with getting the attributes from the grid object, further down the line we might reconsider using global state for the final model. I appreciate you using `rank` for the variable name, the existing model and others seem to pretty consistently use `tile` to refer to one of the 6 cube faces so I've been trying to stick to that nomenclature.

Reply:
We can reconsider global state further down, for sure. For now, it feels wrong to pass the grid around to every single method when almost everything uses it. For these test purposes, you should always be able to use a grid object separate from the savepoint, whether or not it is set as a global state or passed in as an argument. Each translate class gets a grid object as part of its initialization method that is scraped from that first Grid savepoint.

Reply:
Yeah, down the line there will be a lot of other options too like having the grid specifications attached to Quantity so you are passing it around but without including it as a separate argument. It makes sense to use global state while you're doing this porting.